From b37e09c7bf22de12a00b65280181ef11a99de2b3 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Date: Wed, 27 May 2020 12:08:41 +0300
Subject: [PATCH 115/269] s32g: mc_me: Correct pconf and pupb value calculation

This commit fixes a typo in pconf and pupd value calculation.
The original code applies & operator on pconf/pupd register address
instead of applying it on the read value.

Issue: ALB-4907
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/s32g/s32g_mc_me.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/plat/s32g/s32g_mc_me.c b/plat/s32g/s32g_mc_me.c
index 11a0c82ab..4c9f5cc6a 100644
--- a/plat/s32g/s32g_mc_me.c
+++ b/plat/s32g/s32g_mc_me.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019 NXP
+ * Copyright 2019-2020 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -62,8 +62,8 @@ static void mc_me_part_pconf_write_pce(uint32_t pce_bit, uint32_t p)
 {
 	uint32_t pconf;
 
-	pconf = mmio_read_32(S32G_MC_ME_PRTN_N_PCONF(p) &
-			~S32G_MC_ME_PRTN_N_PCONF_PCE_MASK);
+	pconf = mmio_read_32(S32G_MC_ME_PRTN_N_PCONF(p)) &
+			~S32G_MC_ME_PRTN_N_PCONF_PCE_MASK;
 	pconf |= (pce_bit & S32G_MC_ME_PRTN_N_PCONF_PCE_MASK);
 	mmio_write_32(S32G_MC_ME_PRTN_N_PCONF(p), pconf);
 }
-- 
2.17.1

